### ISSUE TITLE
About Query was empty error

### ISSUE THREAD

**Issue by burakkilic** [dchester/epilogue#32]
Hi;

First I wrote this issue to SequelizeJS, but I think the problem can be related with epilogue. 

https://github.com/sequelize/sequelize/issues/2850

I debug the code of epilogue and I think, there can be a bug in the line:

      _(context.attributes).extend(_(req.body).clone());

in update.js

I am making PUT request to my server with data:

    {"id":87,"name":"name1","description":"description1","MenuId":63}

If I submit these values in a form in Postman, it is successful.

If I send raw data, req.body is empty and I am getting this error:

    {"error":{"name":"SequelizeDatabaseError","message":"ER_EMPTY_QUERY: Query was empty","parent":{"code":"ER_EMPTY_QUERY","errno":1065,"sqlState":"42000","index":0,"sql":""},"original":{"code":"ER_EMPTY_QUERY","errno":1065,"sqlState":"42000","index":0,"sql":""},"sql":""}}

**Comment by mbroadst** [comment]
@burakkilic can you show me the code you are using to set up your resource? Also what URL are you posting to? Our tests only cover cases where you post update data in the form to a url like: 

https://localhost:3000/items/<item id> <= {"id":87,"name":"name1","description":"description1","MenuId":63}

POSTing that same data to just https://localhost:3000/items hasn't been tested, although I will test it now

**Comment by mbroadst** [comment]
@burakkilic okay, but what does the actual raw request look like? You're saying if you create the request using postman it works, but it doesn't when using the rest resource with angular. I'm just trying to figure out how to make the test case locally to recreate the error..

**Comment by johngiudici** [comment]
I noticed this issue about a week ago and haven't had a chance to look into it. It seems to be a problem only with sequelize 2 (I have 2.0.0-rc6 installed). When the data in the update isn't any different from the data that already exists, a bad sql statement is generated with an empty data set. I'm not sure what's at fault, but here's a test you can drop in resource.test.js:

```
it('should succeed when updating with no change to the data', function(done) {
      var userData = { username: 'someUser', email: 'someemail@gmail.com' };
      request.post({
        url: test.baseUrl + '/users',
        json: userData
      }, function(error, response, body) {
        expect(response.statusCode).to.equal(201);
        request.put({
          url: test.baseUrl + '/users/' + body.id,
          json: userData
        }, function(err, response, body) {
          expect(response.statusCode).to.equal(200);
          done();
        });
      });
    });
```

**Comment by mbroadst** [comment]
Ah, okay thanks for the test case, and reminder that I should be testing with sequelize2 rather than the 0.3.2 epilogue branch using 1.7.x as the test base.. Turns out it is indeed a sequelize error, I'll sort it out with them.

**Comment by burakkilic** [comment]
Thanks @johngiudici, and thanks @mbroadst

**Comment by mbroadst** [comment]
@burakkilic I'm going to leave this open to remember to fix it :smile:

**Comment by burakkilic** [comment]
Haha, ok:)

**Comment by mbroadst** [comment]
we've now fixed this in sequelize 2.0.0-rc7, @burakkilic can you please confirm?

**Comment by burakkilic** [comment]
Yes, it's ok. Thank you very much!
